misc/utils.py

```diff
@@ -0,0 +1,155 @@
+                "Number of seeds must be 1 (got: " + actorwise_log["active_actors"] + ")"
+            )
+        else:
+            if actors_infected_epoch < actors_infected_total:
+                raise ArithmeticError(
+                    f"Results contradict themselves! \
+                    Number of active actors in {epoch_num} epoch: {actors_infected_epoch} \
+                    number of all actors active so far: {actors_infected_total}"
+                )
+    
+        # update peaks
+        if actorwise_log["active_actors"] > peak_infections_nb:
+            peak_infections_nb = actorwise_log["active_actors"]
+            peak_iteration_nb = epoch_num + 1  # we don't start counting from 0 :)
+        
+        # # update real length of diffusion
+        # if actors_infected_epoch != actors_infected_total:
+        #     simulation_length = epoch_num + 1  # we don't start counting from 0 :)
+
+        # update nb of infected actors
+        actors_infected_total = actors_infected_epoch
+
+    return SimulationResult(
+        actor_id=actor.actor_id,
+        simulation_length=len(detailed_logs) - 1,
+        actors_infected=actors_infected_total,  # do we consider within this number the seed as well?
+        actors_not_infected=actors_nb - actors_infected_total,
+        peak_infections_nb=peak_infections_nb,
+        peak_iteration_nb=peak_iteration_nb
+    )
+
+
+def mean_repeated_results(repeated_results: list[SimulationResult]) -> SimulationResult:
+    rr_dict_all = [asdict(rr) for rr in repeated_results]
+    rr_df_all = pd.DataFrame(rr_dict_all)
+    rr_dict_mean = rr_df_all.groupby("actor_id").mean().reset_index().iloc[0].round(3).to_dict()
+    return SimulationResult(**rr_dict_mean)
+
+
+def save_magrinal_efficiences(marginal_efficiencies: list[SimulationResult], out_path: Path) -> None:
+    me_dict_all = [asdict(me) for me in marginal_efficiencies]
+    pd.DataFrame(me_dict_all).to_csv(out_path)
+
+
+def get_current_time():
+    now = datetime.datetime.now()
+    return now.strftime("%Y-%m-%d %H:%M:%S")
+
+
+def get_diff_of_times(strftime_1, strftime_2):
+    fmt = "%Y-%m-%d %H:%M:%S"
+    t_1 = datetime.datetime.strptime(strftime_1, fmt)
+    t_2 = datetime.datetime.strptime(strftime_2, fmt)
+    return round((t_2 - t_1).seconds / 60, 2)
+
+
+def zip_detailed_logs(logged_dirs: list[Path], rm_logged_dirs: bool = True) -> None:
+    # Ensure at least one directory is provided
+    if len(logged_dirs) == 0:
+        print("No directories provided to create archive from.")
+        return
+    
+    # Get the parent directory of the first directory in logged_dirs
+    parent_dir = logged_dirs[0].parent
+    
+    # Create the name for the zip file based on the parent directory
+    zip_filename = parent_dir / "_output.zip"
+
+    # Create the archive
+    try:
+        logged_dirs_as_str = " ".join([str(ld) for ld in logged_dirs])
+        command = f"zip -r {zip_filename} {logged_dirs_as_str}"
```

Comment:
I really think it's worth the change. Using `shutil` would replace 9 lines of code (with 2 indentations) with 1 short line. Besides the high volume and low readability, current code has other potential issues - e.g. you don't know that the `zip` command is actually available in the os. 

It seems a bit odd that `shutil` would cause problems, but if you prefer you can use [tarfile](https://docs.python.org/3/library/tarfile.html) or [zipfile](https://docs.python.org/3/library/zipfile.html)

Reply:
It caused problems when the archive was supposed to be created recursively, if I recall correctly. Please, let's not dwell into details of this function. It does what it should do, and (I hope) it will not be our biggest contribution :D